lxd/cluster/notify.go

```diff
@@ -84,36 +85,36 @@ func NewNotifier(state *state.State, networkCert *shared.CertInfo, serverCert *s
 			if !HasConnectivity(networkCert, serverCert, member.Address) {
 				switch policy {
 				case NotifyAll:
-					return nil, fmt.Errorf("peer node %s is down", member.Address)
+					return nil, fmt.Errorf("Peer node %s is down", member.Address)
 				case NotifyAlive:
 					continue // Just skip this node
 				case NotifyTryAll:
 				}
 			}
 		}
 
-		peers = append(peers, member.Address)
+		peers = append(peers, member)
 	}
 
-	notifier := func(hook func(lxd.InstanceServer) error) error {
+	notifier := func(hook func(db.NodeInfo, lxd.InstanceServer) error) error {
 		errs := make([]error, len(peers))
 		wg := sync.WaitGroup{}
 		wg.Add(len(peers))
-		for i, address := range peers {
-			logger.Debugf("Notify node %s of state changes", address)
-			go func(i int, address string) {
+		for i, member := range peers {
+			logger.Debugf("Notify node %s at %s of state changes", member.Name, member.Address)
```

Comment:
Notify member...

Reply:
Also this is a good opportunity to move away from `logger.*f()` and use contextual logging